src/UI/Actions.hs

```diff
@@ -388,8 +388,22 @@ removeTags ts =
 reloadList :: Action 'BrowseThreads AppState
 reloadList = Action "reload list of threads" applySearch
 
+selectNextUnread :: Action 'BrowseMail AppState
+selectNextUnread =
+  Action { _aDescription = "select next unread"
+         , _aAction = (\s -> let vec = (view (asMailIndex . miListOfMails . L.listElementsL) s)
+                                 cur = (view (asMailIndex . miListOfMails . L.listSelectedL) s) <|> Just 0
+                                 fx m = Notmuch.isNew (view (asConfig . confNotmuch . nmNewTag) s) m
+                                 next = nextUnread vec cur fx
+                             in pure $ over (asMailIndex . miListOfMails) (L.listMoveTo (fromJust next)) s)
+         }
+
 -- Function definitions for actions
 --
+-- | Return the next unread item or default to the currently selected.
+nextUnread :: Vector.Vector a -> Maybe Int -> (a -> Bool) -> Maybe Int
+nextUnread vec cur fx = Vector.findIndex fx (Vector.drop ((fromJust cur) + 1) vec) <|> cur
```

Comment:
Not very happy about the `fromJust` use here since it could lead to an Exception. However if I simply pattern match the `cur` it should be fine? Perhaps use the `maybe` function?

Reply:
Yeah `maybe` or `fromMaybe`.